src/prefect/cli/deploy.py

```diff
@@ -670,6 +669,13 @@ async def _run_single_deploy(
     deploy_config["parameter_openapi_schema"] = _parameter_schema
     deploy_config["schedules"] = _schedules
 
+    if isinstance(deploy_config.get("concurrency_limit"), dict):
+        concurrency_limit = deploy_config.pop("concurrency_limit")
+        deploy_config["concurrency_options"] = {
+            "collision_strategy": concurrency_limit["collision_strategy"]
+        }
+        deploy_config["concurrency_limit"] = concurrency_limit["limit"]
```

Comment:
It might make sense to use `.get` for dictionary access with default values here for safety.